packages/admin/admin/src/dataGrid/CrudContextMenu.tsx

```diff
@@ -0,0 +1,193 @@
+    );
+};
+
+export interface CrudContextMenuProps<CopyData> {
+    url?: string;
+    onPaste?: (options: { input: CopyData; client: ApolloClient<object> }) => Promise<void>;
+    onDelete?: (options: { client: ApolloClient<object> }) => Promise<void>;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    refetchQueries: RefetchQueriesOptions<any, unknown>["include"];
+    copyData?: () => Promise<CopyData> | CopyData;
+}
+
+export function CrudContextMenu<CopyData>({ url, onPaste, onDelete, refetchQueries, copyData }: CrudContextMenuProps<CopyData>): React.ReactElement {
+    const intl = useIntl();
+    const client = useApolloClient();
+    const errorDialog = useErrorDialog();
+
+    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+    const [deleteDialogOpen, setDeleteDialogOpen] = React.useState(false);
+    const [copyLoading, setCopyLoading] = React.useState(false);
+    const [pasting, setPasting] = React.useState(false);
+
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    const handleDeleteClick = async () => {
+        if (!onDelete) return;
+        await onDelete({
+            client,
+        });
+        await client.refetchQueries({ include: refetchQueries });
+        setDeleteDialogOpen(false);
+    };
+
+    const handlePasteClick = async () => {
+        if (!onPaste) return;
+        const clipboard = await readClipboardText();
+
+        if (clipboard) {
+            let input;
+
+            try {
+                input = JSON.parse(clipboard);
+            } catch (e) {
+                errorDialog?.showError({
+                    userMessage: <FormattedMessage id="comet.common.clipboardInvalidFormat" defaultMessage="Clipboard contains an invalid format" />,
+                    error: e.toString(),
+                });
+                console.error("Bad clidpboard value, parsing JSON failed", e);
+            }
+
+            if (input) {
+                // TODO validate input?
+                try {
+                    await onPaste({
+                        input,
+                        client,
+                    });
+                    await client.refetchQueries({ include: refetchQueries });
+                } catch (e) {
+                    errorDialog?.showError({
+                        userMessage: (
+                            <FormattedMessage
+                                id="comet.common.pasteFailedInvalidFormat"
+                                defaultMessage="Paste failed, probably due to an invalid format"
+                            />
+                        ),
+                        error: e.toString(),
+                    });
+                    console.error("mutation failed", e);
+                }
+            }
+        } else {
+            console.error("Clidpboard is empty");
```

Comment:
```suggestion
```

Reply:
Why don't you show a user error with `errorDialog` here as well?

Reply:
my thought was, that when you paste nothing (in any application), usually simply nothing happens